File: user-service/src/main/java/com/movieapp/user_service/service/UserService.java
package com.movieapp.user_service.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.movieapp.user_service.model.User;

@Service
public interface UserService {
	public List<User> getUsers();
	public User getUserById(String id);
	public User addUser(User user);
	public User updateUser(String id, User user);
	public String deleteUser(String id);
	User getByEmail(String email);
	Boolean emailPresent(String email);

}

File: movie-webapp/src/app/service/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User } from '../model/user.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http:HttpClient) { }

  private userUrl='http://localhost:1111/users'

  getUserById(){
    return this.http.get<User>(this.userUrl)
  }

  // updateUser(user:User){
  //   return this.http.put<User>(this.userUrl,user);
  // }

  addUser(user: User): Observable<User> {
    return this.http.post<User>('http://localhost:1111/auth/register', user);
  }

  deleteUser(id:any){
    return this.http.delete(`${this.userUrl}/${id}`,{responseType: 'text'})
  }

  resetPassword(email:any,answer:any,password:any ): Observable<any> {
    let forget={
      "email": email,
      "answer": answer,
      "newPassword": password
    }
    return this.http.post(`${this.userUrl}/reset-password`, forget,{responseType: 'text'});
  }
  
}
